### PR TITLE
121 bj creating mailers cohort match survey

### LINKED ISSUE #121: TB — Creating Mailers For Cohort, Match, and Survey Models
These mailers will have different actions, listed below.

## Cohort
1. Shortlist start time reached
2. Outlier Email
3. 2 Week Warning/Survey Reminder

## Match
1. Match Created (Mentee and Mentor get email)

## Survey
1. Admin emailed when Survey record created

### PR DESCRIPTION
**Pull request by borvux**
Created cohort, match, and survey mailers. 

My first time working with mailers, I'm anticipating that something might not work as expected, since I wasn't able to test them. I'll need your help, @brvarner , to test these functions.

Also added the association between the Survey and Match models, as suggested by @YoungSong99 

Close #121

### CODE REVIEW

File: app/mailers/cohort_mailer.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  def unmatched_notification(user, cohort)
+    @user = user
+    @cohort = cohort
+    mail(to: @user.email, subject: 'Please Log In to Select a Mentor')
+  end
+  
+  def matching_complete_notification(admin, cohort)
+    @admin = admin
+    @cohort = cohort
+    mail(to: @admin.email, subject: 'Cohort Matching Complete')
```
Comment (MayaS1111) [inline app/mailers/cohort_mailer.rb:17]: Maybe "MentE Matching Complete for #{@cohort.cohort_name}!"

File: app/mailers/cohort_mailer.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+  def two_week_warning(admin_email, cohort)
+    @cohort = cohort
+    mail(to: admin_email, subject: "Two-Week Warning for #{@cohort.cohort_name}")
```
Comment (MayaS1111) [inline app/mailers/cohort_mailer.rb:22]: If this is for the cohort ending say something like: "Two Weeks until #{@cohort.cohort_name} Ends!"
Just so they know what the warning is for.

**Review by MayaS1111 — CHANGES_REQUESTED** [review]
Can you update the name, then your all good!

**Review by MayaS1111 — APPROVED** [review]
Thanks for making the changes, great work Benny!!

**Review by rezahusain — APPROVED** [review]
Looks good Benny! Nice job with the changes.

**Pull request merged**